pkg/cloudclient/cloudclient.go

```diff
@@ -25,22 +55,54 @@ type CloudClient interface {
 	// ValidateEgress validates that all required targets are reachable from the vpcsubnet
 	// target URLs: https://docs.openshift.com/rosa/rosa_getting_started/rosa-aws-prereqs.html#osd-aws-privatelink-firewall-prerequisites
 	// Expected return value is *output.Output that's storing failures, exceptions and errors
-	ValidateEgress(ctx context.Context, vpcSubnetID, cloudImageID string, kmsKeyID string, timeout time.Duration) *output.Output
+	ValidateEgress(ctx context.Context) *output.Output
 
 	// VerifyDns verifies that a given VPC meets the DNS requirements specified in:
 	// https://docs.openshift.com/container-platform/4.10/installing/installing_aws/installing-aws-vpc.html
 	// Expected return value is *output.Output that's storing failures, exceptions and errors
 	VerifyDns(ctx context.Context, vpcID string) *output.Output
 }
 
-func NewClient(ctx context.Context, logger ocmlog.Logger, creds interface{}, region, instanceType string, tags map[string]string) (CloudClient, error) {
-	switch c := creds.(type) {
-	case awscredsv1.Credentials, awscredsv2.StaticCredentialsProvider:
-		return awsCloudClient.NewClient(ctx, logger, c, region, instanceType, tags)
-	case *google.Credentials:
-		return gcpCloudClient.NewClient(ctx, logger, c, region, instanceType, tags)
+func getCloudClientType(options CmdOptions) string {
+	if options.AwsProfile != "" || os.Getenv("AWS_ACCESS_KEY_ID") != "" || options.CloudType == "aws" {
+		return AWS
+	}
+	if options.CloudType == GCP {
+		return GCP
+	}
+	return "unknown"
+}
+
+func NewClient(ctx context.Context, logger ocmlog.Logger,
+	options CmdOptions) (CloudClient, error) {
+
+	logger.Info(ctx, "Using region: %s", getDefaultRegion())
```

Comment:
This prints `us-east-1` regardless of what region is provided

```
AWS_ACCESS_KEY_ID=$ACCESS_KEY AWS_SECRET_ACCESS_KEY=$SECRET_KEY  ./osd-network-verifier egress --subnet-id $SUBNET_ID  --region us-west-2
Using region: us-east-1
Using AWS secret key
```

Reply:
updated

Reply:
I don't see a commit pushed to address this. Possible you forgot to push?

Reply:
Nope. Initially it was printing "options.Region" which will always print what you provide on cli. "getDefaultRegion" function will print the actual region being used. Whether in cli or in the env, or the default val set by code if neither exists.

Reply:
Something is still not right here

```
$  git log | head -1
commit 97abcd959ceb4e285e78bf2e1a67e3b2737b0c3b

$ make build
go fmt ./...
go mod tidy
go build -mod=mod .


$ AWS_ACCESS_KEY_ID=foo AWS_SECRET_ACCESS_KEY=bar ./osd-network-verifier egress --subnet-id subnet-00fdc1eac0be9449d  --region us-west-2
Using region: us-east-1
Using AWS secret key
```